resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/submit/LoggingPodStatusWatcher.scala

```diff
@@ -134,13 +151,6 @@ private[k8s] class LoggingPodStatusWatcherImpl(
     }.mkString("")
   }
 
-  override def awaitCompletion(): Unit = {
-    podCompletedFuture.await()
-    logInfo(pod.map { p =>
-      s"Container final statuses:\n\n${containersDescription(p)}"
-    }.getOrElse("No containers were found in the driver pod."))
-  }
-
```

Comment:
This removal looks like a part of independency PR instead of the part of SPARK-24266. Could you tell us why this is required and where this came from?

Reply:
@shockdm Could you chime in on this one?

Reply:
@dongjoon-hyun It does look like it originates from SPARK-28947 02c5b4f76337cc3901b8741887292bb4478931f3 which eliminated the future and was a rename.

Since this is a private trait, the logic should be completely self-contained and safe to remove.

Reply:
@jkleckner thank you for following up, that is correct. Sorry for the late response :(